ui/calendar.js

```diff
@@ -135,16 +132,26 @@ return $.widget( "ui.calendar", {
 			return;
 		}
 
-		if ( this.date.month() !== oldMonth || this.date.year() !== oldYear ) {
-			this.refresh();
+		if ( this._needsRefresh() ) {
+			this._refresh();
 			this.grid.focus();
 		}
 
 		this._setActiveDescendant();
 	},
 
+	_needsRefresh: function() {
+		if ( this.date.month() !== this.viewDate.month() || this.date.year() !== this.viewDate.year() ) {
+			return !this.grid.find(
```

Comment:
This does a basic checking if anything has changed at all (as before, see https://github.com/fnagel/jquery-ui/commit/4c2229fc51dbb70e1dcb96afdaef92d7c41a44eb#diff-563a5392a566d1a29a68b1b5a37c4cb6L139) and if so, checks if the wanted date is already available within our grid.